### DISCUSSION TITLE
Feature: lerna publish --dist

### DISCUSSION THREAD

**Discussion by Zamiell** [ghiscoding/lerna-lite#202] [General]
Nx is quickly becoming the most popular monorepo tool. However, it does not come with any versioning or publishing tools, and this is where lerna-lite shines, so lerna-lite seems like a natural fit to add on top of an Nx monorepo.

First, let's understand how monorepos are set up with Nx:
At the root of the monorepo, there is a `dist` directory, and all of the packages are compiled to subdirectories there.

```
monorepo/
├── packages/
|   ├── foo
|   └── bar
└── dist/
    └── packages/
        ├── foo
        └── bar
```

However, this directory structure does not seem to be usable with lerna-lite, because it tries to publish from the source directory instead of the output directory. Furthermore, `lerna publish` exposes a `--contents` flag, but I don't think this is helpful for us because the file path to publish is dynamic and based on the package name, so you can't specify it in a simple string.

Since this is a common directory structure, I propose that lerna-lite adds two optional flags to `lerna publish`:
- `--publishBaseDirectory` - The path to the parent directory of the directory to publish. In this case, it would be `monorepo/dist`. Then, when publishing, lerna-lite will `cd` to `publishBaseDirectory/packageName`.
- `--dist` - This would be an alias for `--publishBaseDirectory dist`. A special alias is provided for convenience since this is the most common monorepo/JavaScript/TypeScript convention.

**Comment by ghiscoding** [comment]
Hmm since Lerna is now managed by the Nx team and now also recently switched their own Nx monorepo to Lerna (you can see that in the [Discussion](https://github.com/lerna/lerna/discussions) of their roadmap), so I would rather wait for them to introduce anything related to Nx and they already started with `lerna run` with a PR #183 that I replicated. Basically I'm trying to be on par with original Lerna as much as possible and replicate the same PRs they opened when possible. I also don't know Lerna's codebase enough to do big structure change like you're asking here. I can be open to Pull Request from your side, but sorry to say that you should not expect me to do these changes. 

By the way, I've never seen a structure like you're showing, even though you said it's common (or is it common only for Nx? which I've never used). The most common structure for me is the regular `src` and `dist` folders when using TypeScript
```sh
monorepo/
├── packages/
   ├── foo
       ├── dist
       ├── src
   └── bar
       ├── dist
       ├── src
```
I wonder if the Nx team will work on this now that they took ownership of Lerna and released version 5 recently, I mean they also use Lerna in their own monorepo now. So far it doesn't look like they starting to take a look at all the opened issues/PRs, I would guess that they'll start cleaning them up eventually though.
